gapis/api/vulkan/frame_loop.go

```diff
@@ -781,29 +787,50 @@ func (f *frameLoop) detectChangedResources(ctx context.Context) {
 
 func (f *frameLoop) detectChangedBuffers(ctx context.Context) {
 
-	apiState := GetState(f.loopEndState)
+	apiState := GetState(f.loopStartState)
 
 	// Find out changed buffers.
 	for bufferKey, buffer := range apiState.Buffers().All() {
 
-		data := buffer.Memory().Data()
-		span := interval.U64Span{data.Base(), data.Base() + data.Size()}
-		poolID := data.Pool()
+		toDestroy := f.bufferToDestroy[buffer.VulkanHandle()]
+		toCreate := f.bufferToCreate[buffer.VulkanHandle()]
+
+		if toCreate == true {
+
+			// If we're going to recreate this object for the start of the loop we need to set its state back to the right conditions
+			f.bufferChanged[bufferKey] = true
+			continue
+
+		} else if toDestroy == true {
+
+			// If we created this object during the loop and we're going to destroy this object at the end of the loop then we don't need to capture the state
```

Comment:
You are looping on buffers at f.loopStartState, are you sure that you can list buffers that will be destroyed? (Just making sure the logic holds because you changed loopEndState to loopStartState)